libclc/CMakeLists.txt

```diff
@@ -411,6 +411,16 @@ foreach( t ${LIBCLC_TARGETS_TO_BUILD} )
     set( LIBCLC_ARCH_OBJFILE_DIR "${LIBCLC_OBJFILE_DIR}/${arch_suffix}" )
     file( MAKE_DIRECTORY ${LIBCLC_ARCH_OBJFILE_DIR} )
 
+    # OpenCL 3.0 extensions
+    string(CONCAT CL_3_0_EXTENSIONS
+      "-cl-ext="
+      "+cl_khr_fp64,"
+      "+cl_khr_fp16,"
+      "+__opencl_c_3d_image_writes,"
+      "+__opencl_c_images,"
+      "+cl_khr_3d_image_writes")
+    list( APPEND build_flags -cl-std=CL3.0 "-Xclang" ${CL_3_0_EXTENSIONS} )
```

Comment:
We would need a more target-configurable way of enabling OpenCL extensions. Not all targets do fp16/fp64, not all do 3D image writes, etc.

Reply:
I've reverted OpenCL extensions change in this PR in https://github.com/llvm/llvm-project/pull/135733/commits/4facfec781e39a247aba639ea8e080aa79153a12
I find that target should define its supported extension via setSupportedOpenCLOpts API.